### PR TITLE
feat: improve accessibility for single field components tckt-393

### PR DESCRIPTION
**Pull request by kalasgarov**
Addressed some accessibility issues for single field components.
Improved error handling for single field components.

### CODE REVIEW

File: packages/design/src/Form/components/DateOfBirth/DateOfBirth.tsx [hunk 1]
```diff
@@ -36,20 +37,33 @@ export const DateOfBirthPattern: PatternComponent<DateOfBirthProps> = ({
...
           <select
-            className="usa-select"
+            className={classNames('usa-input', {
+              'usa-input--error': error,
+            })}
             id={monthId}
             {...register(monthId)}
-            aria-describedby="mdHint"
+            aria-describedby={
+              error ? `${monthId} input-error-message-${monthId}}` : monthId
```
Comment (ethangardner) [inline packages/design/src/Form/components/DateOfBirth/DateOfBirth.tsx:65]: I left a comment for you on your other PR about the logic for the `aria-describedby` attribute. The same comment is true here where I think you'll want to have 4 branches of a conditional - error only, hint only, both error and hint, neither error nor hint.

File: packages/design/src/Form/components/DateOfBirth/DateOfBirth.tsx [hunk 2]
```diff
@@ -66,37 +80,40 @@ export const DateOfBirthPattern: PatternComponent<DateOfBirthProps> = ({
...
+            className={classNames('usa-input', {
+              'usa-input--error': error,
+            })}
             id={dayId}
             {...register(dayId)}
             minLength={2}
             maxLength={2}
             pattern="[0-9]*"
             inputMode="numeric"
+            aria-describedby={
```
Comment (ethangardner) [inline packages/design/src/Form/components/DateOfBirth/DateOfBirth.tsx:92]: This is described by `hint-${monthId}` as well. You'll have identical `aria-describedby` logic for all 3 form fields, so you many want to make a function for it.

File: packages/design/src/Form/components/DateOfBirth/DateOfBirth.tsx [hunk 3]
```diff
@@ -66,37 +80,40 @@ export const DateOfBirthPattern: PatternComponent<DateOfBirthProps> = ({
...
+              'usa-input--error': error,
+            })}
             id={yearId}
             {...register(yearId)}
             minLength={4}
             maxLength={4}
             pattern="[0-9]*"
             inputMode="numeric"
+            aria-describedby={
+              error ? `${yearId} input-error-message-${yearId}}}` : yearId
```
Comment (ethangardner) [inline packages/design/src/Form/components/DateOfBirth/DateOfBirth.tsx:112]: Same as my comment above.

File: packages/design/src/Form/components/EmailInput/EmailInput.tsx [hunk 4]
```diff
@@ -18,20 +20,23 @@ export const EmailInputPattern: PatternComponent<EmailInputProps> = ({
...
-          className="usa-input margin-bottom-1"
+          className={classNames('usa-input margin-bottom-1', {
+            'usa-input--error': error,
+          })}
           id={emailId}
           type="email"
           autoCapitalize="off"
           autoCorrect="off"
           {...register(emailId, { required })}
+          aria-describedby={error ? `${emailId} ${errorId}}` : emailId}
```
Comment (ethangardner) [inline packages/design/src/Form/components/EmailInput/EmailInput.tsx:37]: The `emailId` can be safely removed.

File: packages/design/src/Form/components/PhoneNumber/PhoneNumber.tsx [hunk 5]
```diff
@@ -49,7 +46,7 @@ export const PhoneNumberPattern: PatternComponent<PhoneNumberProps> = ({
           type="tel"
           defaultValue={value}
           {...register(phoneId, { required })}
-          aria-describedby="primaryPnHint"
+          aria-describedby={error ? `${phoneId} ${errorId}` : phoneId}
```
Comment (ethangardner) [inline packages/design/src/Form/components/PhoneNumber/PhoneNumber.tsx:49]: Same comment as before with the 4 branches of the conditional.

File: packages/design/src/Form/components/SelectDropdown/SelectDropdown.tsx [hunk 6]
```diff
@@ -12,13 +12,27 @@ export const SelectDropdownPattern: PatternComponent<SelectDropdownProps> = ({
...
+          {error.message}
+        </div>
+      )}
+      <select
+        className={classNames('usa-input', {
+          'usa-input--error': error,
+        })}
+        id={selectId}
+        {...register(selectId, { required })}
+        aria-describedby={error ? `${selectId} ${errorId}` : selectId}
```
Comment (ethangardner) [inline packages/design/src/Form/components/SelectDropdown/SelectDropdown.tsx:34]: `selectId` can be safely removed from this.

**Review by ethangardner — COMMENTED** [review]
I really appreciate you taking the time to do this! I left some comments for you on this one. The changes are minor tweaks to the aria labels, so I don't need to review again and can trust you to make the changes. I'll give it a conditional approval as long as the changes are implemented. Just merge it when the changes are complete.

File: packages/design/src/Form/components/DateOfBirth/DateOfBirth.tsx (hunk 2 above)
Comment (kalasgarov) [reply packages/design/src/Form/components/DateOfBirth/DateOfBirth.tsx]: Good idea! Updated!

**Review by ethangardner — APPROVED** [review]
This looks good to me now. I really appreciate how you saw an opportunity and jumped in to fix it.

I think we probably have a need for some utility functions in the app for UI logic. I can see how the function you wrote for getting the `aria-describedby` ids might be useful in both the form builder and form filler UI. 

I think it should probably live in the `common` package, but this isn't anything to address now. I'm more thinking out loud for the next epic. You're good to merge this to main.

**Pull request merged**